npm/test/auth.test.ts

```diff
@@ -0,0 +1,46 @@
+import { getToken } from "../src/auth"
+jest.mock("keycloak-js")
+
+import Keycloak from "keycloak-js"
+
+class MockKeycloakAuthenticated {
+  token: string = "fake-auth-token"
+
+  init() {
+    return new Promise((resolve, _) => resolve(true))
+  }
+}
+class MockKeycloakUnauthenticated {
+  init() {
+    return new Promise((resolve, _) => resolve(false))
+  }
+}
+class MockKeycloakError {
+  init() {
+    return new Promise((_, reject) => reject("There has been an error"))
+  }
+}
+
+beforeEach(() => jest.resetModules())
+
+test("Returns an error if the user is not logged in", async () => {
+  ;(Keycloak as jest.Mock).mockImplementation(() => {
+    return new MockKeycloakUnauthenticated()
+  })
+  await expect(getToken()).rejects.toEqual("User is not authenticated")
+  jest.clearAllMocks()
```

Comment:
Worth doing this in the before/after each step?

Reply:
This is redundant, the `resetModules` on the beforeEach function does what's needed here. I've removed it.